Extend integration tests

sebwoj opened the pull request:
This commit adds a few new tests to integration tests. Such tests will prevent
issues like #2966. This will increase overall integration test duration, but will
also increase safety.

Signed-off-by: Sebastian Wojciechowski <s.wojciechowski89@gmail.com>

sebwoj commented:
@bowlofeggs I'm not sure about the direction of this PR. My plan is to add new integration tests like:
- GET on every release in DB
- GET on updates query service
- EDIT on update
- CREATE update with bug(test integration with bugzilla if possible)
ans so on.

The list of test cases can be long, so first I would like to know what do you think.

bowlofeggs commented:
+1, we definitely want to add a bunch of integration tests. To get write working, we need to set up an Ipsilon server in our CI suite. @puiterwijk told me that there's a way to configure Ipsilon to have a test user that sounded really easy and handy. So if we configure it that way, it should be pretty straightforward to start testing things like creating updates and editing updates. That would be very nice to have!

sebwoj committed:
Extend integration tests

This commit adds a few new tests to integration tests. Such tests will prevent
issues like #2966. This will increase overall integration test duration, but will
also increase safety.

Signed-off-by: Sebastian Wojciechowski <s.wojciechowski89@gmail.com>

bowlofeggs commented on devel/ci/integration/tests/test_bodhi.py:
```diff
@@ -16,16 +16,471 @@
+                elif value == "logout":
+                    assert "Logout Required" in http_response.text
+            else:
+                assert value in http_response.text
```
Instead of looping over the keys and values, why not assert the keys you expect directly?

sebwoj replied:
I'm not sure I understand. Do you think that it will be better to check for keys in http_response.text instead of values?

bowlofeggs replied:
I meant more to ask why we don't just expect the keys to be present and just make the above assertions, as opposed to the large if/else block. For example, why not do things like this instead:

```
if update_info['request']:
    assert update_info['request'] in http_response.text
assert f"stable threshold: {update_info['stable_karma']}" in http_response.text
<etc>
````

sebwoj replied:
At first, my intention was just to loop over values and check existance(of raw value) in response, but I find out that  most of `update's` fields are rendered different way than just raw value, so I started to adding more and more if/else statements. I will rewrite it.

bowlofeggs commented on devel/ci/integration/tests/test_bodhi.py:
```diff
@@ -16,16 +16,471 @@
+                    assert "Logout Required" in http_response.text
+            else:
+                assert value in http_response.text
+        # ffmarkdown...
```
```suggestion
```

sebwoj replied:
At first I wrote the assertions for comments, but I noticed that we sue "markdown" to create links in the middle of comment text. I decided to drop update's comment tests.

bowlofeggs commented on devel/ci/integration/tests/test_bodhi.py:
```diff
@@ -16,16 +16,471 @@
+            else:
+                assert value in http_response.text
+        # ffmarkdown...
+        # for comment_text in expected_comments:
```
```suggestion
```

bowlofeggs commented on devel/ci/integration/tests/test_bodhi.py:
```diff
@@ -16,16 +16,471 @@
+                assert value in http_response.text
+        # ffmarkdown...
+        # for comment_text in expected_comments:
+        #    assert comment_text in http_response.text
```
```suggestion
```

bowlofeggs reviewed (changes requested):
Looks good so far!

Rather than making one mega pull request, why don't we focus on finishing out the tests in this one, and then add more tests in subsequent PRs? That was they don't become too huge and are easier to review/edit.

bowlofeggs commented on devel/ci/integration/tests/test_bodhi.py:
```diff
@@ -1,4 +1,4 @@
-# Copyright © 2018 Red Hat, Inc.
+# Copyright © 2019 Red Hat, Inc.
```
```suggestion
# Copyright © 2018-2019 Red Hat, Inc.
```

bowlofeggs reviewed (changes requested):
Just need to keep the 2018 and rebase - I'll take care of that for you.

bowlofeggs reviewed (approved):
Thanks!

Commit:
Merge branch 'develop' into 'Extend_integration_tests'

Commit:
Merge branch 'develop' into 'Extend_integration_tests'

Commit:
Merge branch 'develop' into 'Extend_integration_tests'

Commit:
Merge branch 'develop' into 'Extend_integration_tests'

Commit:
Merge branch 'develop' into 'Extend_integration_tests'

Commit:
Merge branch 'develop' into 'Extend_integration_tests'

Commit:
Merge branch 'develop' into 'Extend_integration_tests'

Commit:
Merge branch 'develop' into 'Extend_integration_tests'
